File: eos-process/src/main/java/cn/geosprite/eosprocess/controller/ProcessController.scala
package cn.geosprite.eosprocess.controller

import java.io.{File, IOException}
import java.nio.file.{Files, Paths}

import cn.geosprite.eosprocess.service.BandMathService
import cn.geosprite.eosprocess.utils.Utils.outPath
import javax.servlet.http.HttpServletResponse
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.core.io.FileUrlResource
import org.springframework.http.MediaType
import org.springframework.util.StreamUtils
import org.springframework.web.bind.annotation._

/**
  * @ Author     ：wanghl
  * @ Date       ：Created in 15:34 2019-3-22
  * @ Description：None
  * @ Modified By：
  */
@RestController
class ProcessController {

  @Autowired
  private var service: BandMathService = _

  //controller for ndvi caculating
  @RequestMapping(method = Array(RequestMethod.POST), path = Array("/ndvi"))
  def getNdvi(@RequestParam name: String)= {
    //for example, the name is "L8-Elkton-VA"
    service.getNdvi(name)
  }

  //controller for ndvi .png image generating
  @RequestMapping(value = Array("/png"), method = Array(RequestMethod.GET), produces = Array(MediaType.IMAGE_PNG_VALUE))
  @throws[IOException]
  def getNdviImage(@RequestParam name: String, response: HttpServletResponse): Unit = {
    if (!Files.exists(Paths.get(outPath(name, "png")))) service.getNdvi(name)

    response.setContentType(MediaType.IMAGE_PNG_VALUE)
    val imgFile = new FileUrlResource(outPath(name, "png"))
    //return image to frontend
    StreamUtils.copy(imgFile.getInputStream, response.getOutputStream)

  }
}

File: eos-data/src/main/java/cn/geosprite/eosdata/controller/ProcessController.java
package cn.geosprite.eosdata.controller;

import cn.geosprite.eosdata.dto.OrderStatus;
import cn.geosprite.eosdata.entity.DataGranule;
import cn.geosprite.eosdata.entity.Orders;
import cn.geosprite.eosdata.service.impl.DataServiceImpl;
import cn.geosprite.eosdata.service.impl.ProcessServiceImpl;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.ResponseBody;
import org.springframework.web.bind.annotation.RestController;

import java.sql.Timestamp;
import java.util.List;
import java.util.stream.Collectors;

/**
 * @ Author     ：wanghl
 * @ Date       ：Created in 20:30 2019-4-8
 * @ Description：None
 * @ Modified By：
 */

@RestController
public class ProcessController {

    private ProcessServiceImpl processService;
    private DataServiceImpl dataService;

    @Autowired
    public ProcessController(ProcessServiceImpl processService) {
        this.processService = processService;
    }

    @GetMapping("/process")
    @ResponseBody
    public OrderStatus process(@RequestParam Integer orderId){
        /**
         * 前端发送orderId信息,处理数据，返回处理后的订单状态
         */
     return processService.process(orderId);

    }


    @GetMapping("/test")
    @ResponseBody
    private OrderStatus test(@RequestParam Integer orderId){

        return  new OrderStatus()
                .setMessage("处理完成")
                .setOrderCompletedTime(new Timestamp(System.currentTimeMillis()));

    }
}
